### PR TITLE
[networks] Expire stale entries from `http_in_flight` eBPF map

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

* Add `ebpf.MapCleaner`;
* Expire stale entries from `http_in_flight` eBPF map

### Describe how to test/QA your changes

Start system-probe with the following configuration:
```
system_probe_config:
  log_level: debug
network_config:
  enabled: true
  enable_http_monitoring: true
```

Now in one of window of your terminal, run the following command to spin up a TCP server bound to port 8080:
```
nc -l 8080
```

In another window connect to this server:
```
telnet localhost 8080
```

Once the TCP connection is established type in the following and then hit `<enter>`:
```
GET /foobar HTTP/1.1
```

This will essentially simulate the scenario where a HTTP client issues a request that never gets a response, which would cause an eBPF map entry to "leak".

Now wait 5 minutes and make sure that you see a log message similar to the following

```
2022-06-30 16:27:38 UTC | SYS-PROBE | DEBUG | (pkg/ebpf/map_cleaner.go:126 in clean) | finished cleaning map=Hash(http_in_flight)#61 entries_checked=10 entries_deleted=1 elapsed=209.796µs
```

The important bit of information to validate here is `entries_deleted=1`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/ebpf/map_cleaner.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		}
+	}
+
+	if err := entries.Err(); err != nil {
+		return err
+	}
+
+	elapsed := time.Now().Sub(now)
+	log.Debugf(
+		"finished cleaning map=%s entries_checked=%d entries_deleted=%d, elasped=%s",
```
Comment (brycekahle) [inline pkg/ebpf/map_cleaner.go:128]:
```suggestion
		"finished cleaning map=%s entries_checked=%d entries_deleted=%d, elapsed=%s",
```

File: pkg/ebpf/map_cleaner.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+func (mc *MapCleaner) Stop() {
+	if mc == nil {
+		return
+	}
+
+	mc.stopOnce.Do(func() {
+		mc.done <- struct{}{}
+		close(mc.done)
```
[the remark this answers is not in the document]
Comment (p-lambert) [reply pkg/ebpf/map_cleaner.go:103]: sort of. that would generate a "receive" event on the other side, but I'm using the send to actually _synchronize_ with the other go routine. that way `Stop()` blocks until the go-routine has finished traversing the map

[the remark this answers is not in the document]
Comment (brycekahle) [reply pkg/ebpf/map_cleaner.go:103]: Got it. Maybe add a comment so we know that it is intentional?

[the remark this answers is not in the document]
Comment (p-lambert) [reply pkg/ebpf/map_cleaner.go:103]: sounds good!

File: pkg/network/http/ebpf_main.go [hunk 3]
```diff
@@ -45,6 +46,8 @@ const (
 	batchNotificationsChanSize = 100
 
 	probeUID = "http"
+
+	maxRequestLinger = 30 * time.Second
```
Comment (brycekahle) [inline pkg/network/http/ebpf_main.go:50]: I don't remember the exact semantics in eBPF, but is there any danger of removing an entry that might get matched up when the next `/connections` query is done? Like do we need to wait at least `2*x` (where `x` is connection interval time), to ensure request/responses from two adjacent intervals have a chance to match?

File: pkg/ebpf/map_cleaner.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+	mc.once.Do(func() {
+		ticker := time.NewTicker(interval)
+		go func() {
+			defer ticker.Stop()
+
+			for {
+				select {
+				case <-ticker.C:
+					// TODO: use pkg/security/probe/time_resolver.go to account for
```
[the remark this answers is not in the document]
Comment (p-lambert) [reply pkg/ebpf/map_cleaner.go:83]: I think you're right. From what I read `unix.CLOCK_MONOTONIC` will also give us the uptime _without_ accounting for the suspended time

**Review by brycekahle — APPROVED** [review]

**Comment by nplanel** [comment]
@p-lambert I think we should implement this in the kernel space (triggered by the userspace C.ticker)
We could use a syscall (ioctl) like CWS team do, to call our cleanup maintenance/check in the kernel, would be faster to cleanup the in flight map and less complicated.

**Comment by p-lambert** [comment]
@nplanel that definitely sounds a good idea to me, but I would approach it as a future improvement over the current change and work it on a separate PR if we determine that we need something more performant. for this is supposed to run once every 5 min and I suspect it should be "good enough", but happy to revisit this once we have some data :)

**Comment by nplanel** [comment]
> @nplanel that definitely sounds a good idea to me, but I would approach it as a future improvement over the current change and work it on a separate PR if we determine that we need something more performant. for now this is supposed to run once every 5 min and I suspect it should be "good enough", but happy to revisit this once we have some data :)

Agree every 5min scan, should be a small impact. Let's see in future PR.

**Review by nplanel — APPROVED** [review]

**Pull request merged**